### PR TITLE
ref(typing): fix all strong types

### PR DESCRIPTION
**Pull request by anonrig**
The definitions for stronger types are wrong. It should not start with `src.` prefix.

Pinging @armenzg for visibility 

cc @getsentry/python-typing

### CODE REVIEW

File: src/sentry/snuba/metrics/extraction.py [hunk 1]
```diff
@@ -565,22 +565,22 @@ class SupportedBy:
     on_demand_metrics: bool
 
     @classmethod
-    def neither(cls):
+    def neither(cls) -> SupportedBy:
```
Comment (asottile-sentry) [inline src/sentry/snuba/metrics/extraction.py:568]: `-> Self` is slightly more accurate (imported from `typing`)

File: src/sentry/snuba/metrics/extraction.py [hunk 2]
```diff
@@ -565,22 +565,22 @@ class SupportedBy:
...
         return cls(standard_metrics=False, on_demand_metrics=False)
 
     @classmethod
-    def both(cls):
+    def both(cls) -> SupportedBy:
         return cls(standard_metrics=True, on_demand_metrics=True)
 
     @classmethod
-    def combine(cls, *supported_by):
+    def combine(cls, *supported_by: Any) -> SupportedBy:
```
Comment (asottile-sentry) [inline src/sentry/snuba/metrics/extraction.py:576]: I suspect the `Any` here is `SupportedBy` ?

File: src/sentry/snuba/metrics/extraction.py (hunk 1 above)
Comment (anonrig) [reply src/sentry/snuba/metrics/extraction.py]: Ah nice. Didn't know it was a thing.

**Review by asottile-sentry — APPROVED** [review]

**Pull request merged**